infrastructure/cli/lib/table.ts

```diff
@@ -8,18 +8,42 @@ type TableOptions = {
   borders?: boolean
 }
 
+function coerceToString(value: any): string {
+  if (typeof value !== 'string') {
+    return String(value ?? '')
+  }
+  return value
+}
+
 function* table(
   rows: Row[],
   columns: ColumnInput[],
   {borders = true}: TableOptions = {},
 ): Generator<string, void, void> {
+  // force row values to be strings
+  rows.forEach(row => {
+    Object.keys(row).forEach(key => {
```

Comment:
I have a slight preference for a `for` loop over`Object.entries`, e.g.

```JavaScript
for (let [key, item] of Object.entries(row)) {
  // [...]
}
```

but it's kinda the same thing ☺️

Reply:
i agree that would be much clearer to read. what i was aiming to avoid (perhaps incorrectly?) is a concurrent modification problem: i modify `row` as i iterate over its entries. i was hoping that `Object.keys(row)` returning a new array would allow me to iterate over row keys and concurrently modify the row object itself.

as you point out `Object.entries(row)` also seems to return a new array (rather than a "view" over the row object) so i think i'll go with your idea as it's more readable

Reply:
I think the object properties are really like pointers to where the values are, e.g.

```
// row
{
  [------]         [--------]
  | key1 --------> | value1 |
  [------]         [--------]

  // ...
}
```
so in this specific loop, you're updating `row[key]` with a new value and that should be fine, the location of the object itself shouldn't matter as we're not adding a bunch of new property/values forcing JS to move the object somewhere else in the heap or something...

I _**think**_ the `for (... of Object.entries())` should work in this case but I don't know if there is some of the usual JS weirdness causing this to not be the case.

Don't worry to much about it, if you want to leave it as it is it's also fine, as I said it's that I usually fine `for (... of)` easier to process ☺️

Reply:
Besides even IF javascript move the whole `row` object instance somewhere else for whatever reason (which I don't think it would anyway) `obj[row]` would still point to the right thing/work.

But I appreciate your cautiousness when it comes to loops updating data structures because they can be sneaky.